partiql-eval/src/main/kotlin/org/partiql/lang/eval/internal/IonValueToExprValue.kt

```diff
@@ -0,0 +1,91 @@
+package org.partiql.lang.eval.internal
+
+import com.amazon.ion.IonBlob
+import com.amazon.ion.IonBool
+import com.amazon.ion.IonClob
+import com.amazon.ion.IonDatagram
+import com.amazon.ion.IonDecimal
+import com.amazon.ion.IonFloat
+import com.amazon.ion.IonInt
+import com.amazon.ion.IonList
+import com.amazon.ion.IonSexp
+import com.amazon.ion.IonString
+import com.amazon.ion.IonStruct
+import com.amazon.ion.IonSymbol
+import com.amazon.ion.IonTimestamp
+import com.amazon.ion.IonValue
+import org.partiql.lang.eval.BAG_ANNOTATION
+import org.partiql.lang.eval.Bindings
+import org.partiql.lang.eval.DATE_ANNOTATION
+import org.partiql.lang.eval.ExprValue
+import org.partiql.lang.eval.GRAPH_ANNOTATION
+import org.partiql.lang.eval.MISSING_ANNOTATION
+import org.partiql.lang.eval.TIME_ANNOTATION
+import org.partiql.lang.eval.internal.ext.namedValue
+import org.partiql.lang.eval.time.Time
+import org.partiql.lang.graph.ExternalGraphReader
+import org.partiql.lang.util.bytesValue
+import java.math.BigDecimal
+
+/**
+ * Creates a new [ExprValue] instance from any Ion value.
+ *
+ * If possible, prefer the use of the other methods instead because they might return [ExprValue] instances
+ * that are better optimized for their specific data type (depending on implementation).
+ */
+internal fun ionValueToExprValue(value: IonValue): ExprValue {
```

Comment:
Ported from `ExprValue.kt` and renamed from `ExprValue.of(value)` to `ionValueToExprValue(value)`